File: packages/sdui/src/main/java/nl/q42/sdui/component/TextComponent.java
package nl.q42.sdui.component;

import nl.q42.common.components.ComponentTypes;
import nl.q42.common.components.TextColor;
import nl.q42.common.components.TextFormatting;

public class TextComponent extends Component
{

  public TextColor      color;
  public TextFormatting formatting;
  public String         text;

  public TextComponent(
      String text,
      TextColor color,
      TextFormatting formatting,

      String contentId)
  {
    super(ComponentTypes.TEXT, contentId);
    this.formatting = formatting;
    this.color      = color;
    this.text       = text;
  }

  public TextComponent(String text, String contentId)
  {
    this(text, TextColor.PRIMARY, TextFormatting.NORMAL, contentId);
  }
}

File: packages/sdui-common/src/commonMain/kotlin/nl/q42/common/components/TextComponent.kt
package nl.q42.common.components

import kotlinx.serialization.*

@Serializable
@SerialName(ComponentTypes.TEXT)
class TextComponent(
    val text: String,
    val color: TextColor,
    var size: TextSize,
    val formatting: TextFormatting,
    override val contentId: String
) : Component(ComponentTypes.TEXT) {
    constructor(
        text: String,
        contentId: String,
        color: TextColor = TextColor.PRIMARY,
        size: TextSize = TextSize.NORMAL,
        formatting: TextFormatting = TextFormatting.NORMAL
    ) : this(text, color, size, formatting, contentId)
}

enum class TextColor {
    PRIMARY,
    SECONDARY,
    TERTIARY,
    DISABLED,
    ERROR,
    SUCCESS,
    INFO,
    CUSTOM;
}

enum class TextFormatting {
    NORMAL,
    BOLD,
    ITALIC,
    UNDERLINE;
}

enum class TextSize {
    EXTRA_SMALL,
    SMALL,
    NORMAL,
    LARGE,
    EXTRA_LARGE,
}